src/main/java/org/radarcns/service/SourceMonitorService.java

```diff
@@ -0,0 +1,179 @@
+ * limitations under the License.
+ */
+
+package org.radarcns.service;
+
+import static org.radarcns.mongo.util.MongoHelper.DESCENDING;
+
+import com.mongodb.MongoClient;
+import com.mongodb.client.MongoCursor;
+import java.net.ConnectException;
+import javax.inject.Inject;
+import org.bson.Document;
+import org.radarcns.domain.managementportal.SourceType;
+import org.radarcns.domain.restapi.States;
+import org.radarcns.domain.restapi.header.EffectiveTimeFrame;
+import org.radarcns.mongo.util.MongoHelper;
+import org.radarcns.util.RadarConverter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Generic sourceType monitor.
+ */
+public class SourceMonitorService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SourceMonitorService.class);
+    private static final String TIME_START ="timeStart";
+    private static final String TIME_END ="timeEnd";
+
+    private final MongoClient mongoClient;
+
+
+    /** Constructor. **/
+    @Inject
+    public SourceMonitorService(MongoClient mongoClient) {
+        this.mongoClient = mongoClient;
+    }
+
+
+    public EffectiveTimeFrame getEffectiveTimeFrame(String projectId, String subjectId,
+            String sourceId, SourceType sourceType) {
+
+        // get the last document sorted by timeEnd
+        MongoCursor<Document> cursor = MongoHelper.findDocumentByProjectAndSubjectAndSource
+                (projectId, subjectId, sourceId,
+                        TIME_END, DESCENDING, 1,
+                        MongoHelper.getCollection(this.mongoClient,
+                                sourceType.getSourceStatisticsMonitorTopic()));
+
+        if (!cursor.hasNext()) {
+            LOGGER.debug("Empty cursor for collection {}",
+                    sourceType.getSourceStatisticsMonitorTopic());
+        }
+        long timeStart = Long.MIN_VALUE;
+        long timeEnd = Long.MAX_VALUE;
+        if (cursor.hasNext()) {
+            Document document = cursor.next();
+            timeStart = Math.max(timeStart, document.getDate(TIME_START).getTime());
+            timeEnd = Math.min(timeEnd, document.getDate(TIME_END).getTime());
+        }
+
+        cursor.close();
+        return new EffectiveTimeFrame(RadarConverter.getISO8601(timeStart), RadarConverter
+                .getISO8601(timeEnd));
+
+    }
+    /**
+     * Checks the status for the given sourceType counting the number of received messages and
+     *      checking whether it respects the data frequencies. There is a check for each data.
+     *
+     * @param subjectId identifier
+     * @param sourceId identifier
+     * @param client is the MongoDB client
+     * @return {@code SourceDefinition} representing a sourceType sourceType
+     * @throws ConnectException if the connection with MongoDb is faulty
+     *
+     * @see Source
+     */
+//    public Source getState(String subjectId, String sourceId, MongoClient client ,  double countTemp)
```

Comment:
please uncomment method definition, adding todo's for possible implementations and possibly `UnsupportedOperationException`.